### PR TITLE
🐛 Use `arduino/setup-protoc` for installing Protoc

### PR DESCRIPTION
**Pull request by azeemshaikh38**
* **Please check if the PR fulfills these requirements**
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] PR title follows the guidelines defined in  https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Use `arduino/setup-protoc` action to install protoc. Its way faster than doing it ourselves.

Note: if we choose to do protoc installation ourselves in the future, we should simply download the release archive which will have the `protoc` binary.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -16,32 +16,11 @@ name: build
...
-         ./configure
-         make
-         make check
-     - name: Install protobuf library
-       run: |
-         cd protobuf
-         sudo make install
-         sudo ldconfig
+     - name: Install Protoc
+       uses: arduino/setup-protoc@v1
```
Comment (naveensrinivasan) [inline .github/workflows/main.yml:23]: Could we pin the  SHA instead of `@v1`?

**Review by naveensrinivasan — APPROVED** [review]
Thank you!

File: .github/workflows/main.yml (hunk 1 above)
Comment (azeemshaikh38) [reply .github/workflows/main.yml:23]: Apologies, auto-merge was enabled so couldn't get to this in time. However, I was not able to find the release SHA. Do you know where I can find it?

Comment (naveensrinivasan) [reply .github/workflows/main.yml:23]: Not an issue  👍 
This is the location for the SHA's https://github.com/arduino/setup-protoc/releases and https://github.com/arduino/setup-protoc/commit/64c0c85d18e984422218383b81c52f8b077404d3

Comment (azeemshaikh38) [reply .github/workflows/main.yml:23]: Ah I see, so for GitHub releases we pin using commit SHA?

Comment (naveensrinivasan) [reply .github/workflows/main.yml:23]: Yes!

> Ah I see, so for GitHub releases we pin using commit SHA?

**Pull request merged**